src/Microsoft.TestPlatform.CommunicationUtilities/SocketCommunicationManager.cs

```diff
@@ -119,12 +118,13 @@ public IPEndPoint HostServer(IPEndPoint endpoint)
 
                 var client = await this.tcpListener.AcceptTcpClientAsync();
                 this.socket = client.Client;
-                this.stream = client.GetStream();
-                this.binaryReader = new BinaryReader(this.stream);
+                this.socket.NoDelay = true;
+                this.stream = new PlatformStream().PlatformBufferedStream(client.GetStream(), SocketConstants.BUFFERSIZE);
```

Comment:
Add comment.

Reply:
Got it, as we do socket.poll here also. Need to do in Socket.Client and Socket.Server.

---
In reply to: [137203711](https://github.com/Microsoft/vstest/pull/1041#discussion_r137203711) [](ancestors = 137203711)